### PR TITLE
frontend: add tooltip to core design & project selector

### PR DESCRIPTION
**Pull request by scarlettperry**
### Description
PR
* adds tooltip to core design system
* adds tooltip stories
* adds tooltip to project selector component 
* makes small change to typography component to accept a color prop

### Testing Performed
locally and storybook

Screen Shot 2021-07-28 at 5 15 51 PM

Screen Shot 2021-07-28 at 5 28 50 PM

Screen Shot 2021-07-28 at 5 28 35 PM

### TODOs
- [x] styling for distance b/w tooltip and reference element

### CODE REVIEW

File: frontend/packages/core/src/typography.tsx [hunk 1]
```diff
@@ -118,25 +118,26 @@ type TextVariant =
...
 
 export interface TypographyProps {
   variant: TextVariant;
   children: React.ReactNode;
+  color?: string;
 }
 
-const Typography = ({ variant, children }: TypographyProps) => {
-  return <StyledTypography variant={variant}>{children}</StyledTypography>;
+const Typography = ({ variant, children, color = "#0D1030" }: TypographyProps) => {
```
Comment (scarlettperry) [inline frontend/packages/core/src/typography.tsx:135]: i could have just inline styled the font color in the import component but I think users would want to be able to change the font color, so I added a color prop

File: frontend/packages/core/src/Feedback/tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+import styled from "@emotion/styled";
+import type { TooltipProps as MuiTooltipProps } from "@material-ui/core";
+import { Tooltip as MuiTooltip } from "@material-ui/core";
+
+const BaseTooltip = ({ className, ...props }: MuiTooltipProps) => (
```
Comment (scarlettperry) [inline frontend/packages/core/src/Feedback/tooltip.tsx:6]: the tooltip component accepts a popper component, so initially I was styling the tooltip by targeting it through the popper. however, things got complicated when I wanted to make the maxWidth prop-based as  I'd have to pass the prop from tooltip --> popper component. I was getting various errors and it felt messy. 

Found this pattern here https://stackoverflow.com/questions/59934683/style-material-ui-tooltip-using-emotion-styled and we also do something similar for our Select component https://github.com/lyft/clutch/blob/main/frontend/packages/core/src/Input/select.tsx#L61-L78

File: frontend/packages/core/src/Feedback/tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  children: React.ReactElement;
+  // material ui default is 300px
+  maxWidth?: string;
+  // tooltip text
+  title: React.ReactNode;
+}
+
+const Tooltip = ({ children, maxWidth = "300px", title, ...props }: TooltipProps) => {
+  return (
+    <StyledTooltip title={title} data-max-width={maxWidth} {...props}>
```
Comment (scarlettperry) [inline frontend/packages/core/src/Feedback/tooltip.tsx:43]: title prop supports html https://material-ui.com/components/tooltips/#customized-tooltips

File: frontend/workflows/projectSelector/src/project-selector.tsx [hunk 4]
```diff
@@ -171,9 +177,42 @@ const ProjectSelector = () => {
...
+                    <Typography variant="body3" color="#E7E7EA">
+                      {downstreamDescription}
+                    </Typography>
+                  </TooltipContainer>
+                </>
+              }
+              placement="right-start"
+              maxWidth="400px"
+            >
+              <InfoOutlinedIcon />
```
Comment (danielhochman) [inline frontend/workflows/projectSelector/src/project-selector.tsx:214]: icon is too large

File: frontend/workflows/projectSelector/src/project-selector.tsx [hunk 5]
```diff
@@ -171,9 +177,42 @@ const ProjectSelector = () => {
...
+              title={
+                <>
+                  <TooltipContainer>
+                    <Typography variant="subtitle3" color="#FFFFFF">
+                      Projects
+                    </Typography>
+                    <Typography variant="body3" color="#E7E7EA">
+                      {projectDescription}
+                    </Typography>
+                  </TooltipContainer>
```
Comment (danielhochman) [inline frontend/workflows/projectSelector/src/project-selector.tsx:192]: Actually I'd probably turn this into a loop

**Review by danielhochman — COMMENTED** [review]
Looks great! A few comments:
- The tooltip needs to be offset vertically a bit upwards. The top is not be directly inline with the icon in the design.
- The tooltip should not disappear when moving the mouse to it so it's possible to select text, click a link or a button within the tooltip.

File: frontend/packages/core/src/Feedback/tooltip.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (scarlettperry) [reply frontend/packages/core/src/Feedback/tooltip.tsx]: thxs, updated in typography component too

File: frontend/workflows/projectSelector/src/project-selector.tsx (hunk 4 above)
Comment (scarlettperry) [reply frontend/workflows/projectSelector/src/project-selector.tsx]: from the designs, the 2 icons seem to be the same size so I made them both smaller
Screen Shot 2021-07-30 at 12 56 55 PM

The design specifies 18px but it looks too small IMO
Screen Shot 2021-07-30 at 12 50 46 PM

I brought them down to `fontSize=small` instead which is 20px. LMK what you think
Screen Shot 2021-07-30 at 12 50 54 PM

File: frontend/workflows/projectSelector/src/project-selector.tsx (hunk 5 above)
Comment (scarlettperry) [reply frontend/workflows/projectSelector/src/project-selector.tsx]: turned into a loop

**Comment by scarlettperry** [comment]
@danielhochman  RE 

> The tooltip needs to be offset vertically a bit upwards. The top is not be directly inline with the icon in the design.

I have margin styling w.r.t to how close a tooltip should be to it's reference element but since the tooltip has 12 placement choices https://material-ui.com/components/tooltips/#positioned-tooltips, I've debated making additional changes to where the tooltip should lie since the Figmas don't cover all the placement options. Alternatively though, we can limit the placmement options to just `top`, `bottom`, `left`, and `right`. Wdyt?

File: frontend/packages/core/src/Feedback/tooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  "&.MuiTooltip-tooltipPlacementTop": {
+    margin: "2px 0",
+  },
+  "&.MuiTooltip-tooltipPlacementBottom": {
+    margin: "2px 0",
+  },
+}));
+
+export interface TooltipProps extends Pick<MuiTooltipProps, "interactive" | "placement"> {
```
Comment (scarlettperry) [inline frontend/packages/core/src/Feedback/tooltip.tsx:28]: fyi `interactive` prop is for making the tooltip not disappear when moving the mouse to it

**Review by danielhochman — APPROVED** [review]
we can do the margin fix in TODO and clarify positioning with kanchan

File: frontend/packages/core/src/typography.tsx [hunk 7]
```diff
@@ -118,25 +118,26 @@ type TextVariant =
...
-    fontSize: `${STYLE_MAP[props?.variant || 0].size}px`,
-    fontWeight: STYLE_MAP[props?.variant || 0].weight,
-    lineHeight: `${STYLE_MAP[props?.variant || 0].lineHeight}px`,
-    ...STYLE_MAP[props?.variant || 0].props,
-  })
-);
+const StyledTypography = styled.div<{ color?: string; variant: TextVariant }>(props => ({
+  color: props.color,
+  fontSize: `${STYLE_MAP[props?.variant || 0].size}px`,
+  fontWeight: STYLE_MAP[props?.variant || 0].weight,
```
[the remark this answers is not in the document]
Comment (scarlettperry) [reply frontend/packages/core/src/typography.tsx:124]: yeah the `||` shouldn't be necessary. Will open a sep PR now

[the remark this answers is not in the document]
Comment (scarlettperry) [reply frontend/packages/core/src/typography.tsx:124]: oh actually, in looking at the code more closely, maybe it makes sense to keep the `||` case. It's not possible to set the prop to an undefined variant but if someone creates a new one (ie `caption4`) and doesn't define some of these css properties, then it's value is `undefinedpx` (tested in the browser). From local testing, it doesn't break anything but I think @dschaller added it as a safety check

**Pull request merged**